File: node/src/tools/aube.mjs
import { aubeConfigPath } from "../util/paths.mjs";
import { readSafe, writeAtomic } from "../util/io.mjs";
import { readKey, setKey, removeKey } from "../util/lines.mjs";
import { readExtras, applyExtras, removeExtras } from "../util/extras.mjs";

export const name = "aube";
export const key = "minimumReleaseAge";
export const docs = "https://aube.jdx.dev/";
export const extras = [
  { key: "paranoid", expected: "true", line: "paranoid = true" },
];

export function path(ctx) { return aubeConfigPath(ctx.env, ctx.home, ctx.platform); }

function parseDays(value) {
  if (value === null) return null;
  const n = Number(value);
  if (!Number.isFinite(n) || n <= 0) return null;
  return Math.floor(n / 1440);
}

export async function read(ctx) {
  const p = path(ctx);
  const raw = await readSafe(p);
  const value = readKey(raw, key);
  return { path: p, configured: value, days: parseDays(value), extras: readExtras(raw, extras) };
}

export async function write(days, ctx) {
  const p = path(ctx);
  let text = setKey(await readSafe(p), key, `${key} = ${days * 1440}`);
  text = applyExtras(text, extras);
  await writeAtomic(p, text);
  return { path: p };
}

export async function unset(ctx) {
  const p = path(ctx);
  const before = await readSafe(p);
  const cooldown = removeKey(before, key);
  const ex = removeExtras(cooldown.text, extras);
  const removed = cooldown.removed || ex.removed;
  if (removed) await writeAtomic(p, ex.text);
  return { path: p, removed };
}

File: python/src/pmsec/tools/aube.py
from __future__ import annotations

from pathlib import Path

from pmsec.util.context import Context
from pmsec.util.extras import apply_extras, read_extras, remove_extras
from pmsec.util.io import write_atomic
from pmsec.util.lines import read_key, remove_key, set_key
from pmsec.util.paths import aube_config_path

NAME = "aube"
KEY = "minimumReleaseAge"
DOCS = "https://aube.jdx.dev/"
EXTRAS = [
    {"key": "paranoid", "expected": "true", "line": "paranoid = true"},
]


def path(ctx: Context) -> Path:
    return aube_config_path(ctx.env, ctx.home, ctx.platform)


def _parse_days(value: str | None) -> int | None:
    if value is None:
        return None
    try:
        n = int(value)
    except ValueError:
        return None
    if n <= 0:
        return None
    return round(n / 1440)


def read(ctx: Context) -> dict:
    p = path(ctx)
    raw = p.read_text("utf-8") if p.exists() else ""
    value = read_key(raw, KEY)
    return {"path": str(p), "configured": value, "days": _parse_days(value), "extras": read_extras(raw, EXTRAS)}


def write(days: int, ctx: Context) -> dict:
    p = path(ctx)
    raw = p.read_text("utf-8") if p.exists() else ""
    text = apply_extras(set_key(raw, KEY, f"{KEY} = {days * 1440}"), EXTRAS)
    write_atomic(p, text)
    return {"path": str(p)}


def unset(ctx: Context) -> dict:
    p = path(ctx)
    if not p.exists():
        return {"path": str(p), "removed": False}
    before = p.read_text("utf-8")
    after, removed_cooldown = remove_key(before, KEY)
    after, removed_extras = remove_extras(after, EXTRAS)
    removed = removed_cooldown or removed_extras
    if removed:
        write_atomic(p, after)
    return {"path": str(p), "removed": removed}
